atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/async/AsyncClusterSessionImpl.java

```diff
@@ -0,0 +1,135 @@
+    private final String sessionName;
+    private final Executor executor;
+
+    private final ScheduledExecutorService service = PTExecutors.newSingleThreadScheduledExecutor(
+            new NamedThreadFactory(sessionName() + "-healtcheck", true /* daemon */));
+    private final PreparedStatement healthCheckStatement;
+
+    public static AsyncClusterSessionImpl create(String clusterName, CassandraClusterSessionPair pair,
+            ThreadFactory threadFactory) {
+        return create(clusterName, pair, Executors.newCachedThreadPool(threadFactory));
+    }
+
+    public static AsyncClusterSessionImpl create(String clusterName, CassandraClusterSessionPair pair,
+            Executor executor) {
+        PreparedStatement healthCheckStatement = pair.session().prepare("SELECT dateof(now()) FROM system.local ;");
+        return new AsyncClusterSessionImpl(clusterName, pair, executor, healthCheckStatement);
+    }
+
+    private AsyncClusterSessionImpl(String sessionName, CassandraClusterSessionPair pair, Executor executor,
+            PreparedStatement healthCheckStatement) {
+        this.sessionName = sessionName;
+        this.pair = pair;
+        this.executor = executor;
+        this.healthCheckStatement = healthCheckStatement;
+    }
+
+    @Override
+    public Map<MetricName, Metric> usedCqlClusterMetrics() {
+        return KeyedStream.stream(
+                pair.cluster().getMetrics()
+                        .getRegistry()
+                        .getMetrics())
+                .mapKeys(name -> MetricName.builder().safeName(name).build())
+                .collectToMap();
+    }
+
+    @Nonnull
+    @Override
+    public String sessionName() {
+        return sessionName;
+    }
+
+    @Override
+    public void close() {
+        service.shutdownNow();
+        log.info("Shutting down health checker for cluster session {}", SafeArg.of("clusterSession", sessionName));
+        boolean shutdown = false;
+        try {
+            shutdown = service.awaitTermination(5, TimeUnit.SECONDS);
+        } catch (InterruptedException e) {
+            log.error("Interrupted while shutting down health checker. Should not happen");
+            Thread.currentThread().interrupt();
+        } finally {
+            AsyncSessionManager.getOrInitializeAsyncSessionManager().closeClusterSession(this);
+        }
+
+        if (!shutdown) {
+            log.error("Failed to shutdown health checker in a timely manner for {}",
+                    SafeArg.of("clusterSession", sessionName));
+        } else {
+            log.info("Shut down health checker for cluster session {}", SafeArg.of("clusterSession", sessionName));
+        }
+    }
+
+    @Override
+    public  void start() {
+        service.scheduleAtFixedRate(() -> {
+            ListenableFuture<String> time = getTimeAsync();
+            try {
+                log.info("Current cluster time is: {}", SafeArg.of("clusterTime", time.get()));
```

Comment:
is there another way of indicating failure? probably in a follow up PR, but it would be good to  have some sort of object/sink that exposes whether or not async calls have been failing recently and the result of this should then be fed into that sync. But we can discuss this outside this PR.